hl_extractor/hl_calc.py

```diff
@@ -131,85 +111,56 @@ def create_profile(in_file, out_file, sha1):
 def get_build_sha1(binary):
     """Calculate the SHA1 of the binary we're using."""
     try:
-        f = open(binary, "r")
-        bin = f.read()
-        f.close()
+        with open(binary, "rb") as fp:
+            contents = fp.read()
     except IOError as e:
         print("Cannot calculate the SHA1 of the high-level extractor binary: %s" % e)
         sys.exit(-1)
 
-    return sha1(bin).hexdigest()
+    return hashlib.sha1(contents).hexdigest()
+
+
+def save_hl(mbid, rowid, hl_data, build_sha1):
+    try:
+        jdata = json.loads(hl_data)
+    except ValueError:
+        print("error %s: Cannot parse result document" % mbid)
+        print(hl_data)
+        jdata = {}
 
+    db.data.write_high_level(mbid, rowid, jdata, build_sha1)
 
-def main(num_threads):
+
+def cf_main(num_threads):
```

Comment:
We should need to change the name where this gets invoked right: https://github.com/metabrainz/acousticbrainz-server/blob/0cfc2a72/worker_manage.py#L17?